lua/entities/hl_camera/init.lua

```diff
@@ -128,13 +129,66 @@ function ENT:SetPlayerKey(ply, key, toggle)
 	self:UpdatePlayer(ply)
 end
 
+function ENT:GhostCreate()
+	if IsValid(self.ACGhost) then
+		return
+	end
+	local ghost = ents.Create("hl_camera_ghost")
+
+	ghost:Spawn()
+
+	ghost:SetParent(self)
+	ghost:SetLocalPos(self:GetViewOffset())
+	ghost:SetLocalAngles(Angle(0, 0, self:GetRoll()))
+
+	return ghost
+end
+
+function ENT:UpdateGhostVar(varchanged, oldvalue, newvalue)
+	if oldvalue == newvalue then return end
+
+	if varchanged == "EnableGhost" then
+		self:SetGhost(newvalue)
+	elseif varchanged == "ViewOffset" then
+		self:SetGhostOffset(newvalue, self:GetRoll(), self:GetEnableGhost())
+	elseif varchanged == "Roll" then
+		self:SetGhostOffset(self:GetViewOffset(), newvalue, self:GetEnableGhost())
+	end
+
+end
+
+function ENT:SetGhost(enabled)
+	if !IsValid(self.ACGhost) and enabled then
+		self.ACGhost = self:GhostCreate()
+	elseif !enabled and IsValid(self.ACGhost) then
+		self.ACGhost:Remove()
+	end
+end
+
+function ENT:SetGhostOffset(offset, roll, enabled)
+	if offset == vector_origin then
+		if IsValid(self.ACGhost) then
+			self.ACGhost:Remove()
+		end
+	elseif enabled then
+		if !IsValid(self.ACGhost) then
+			self.ACGhost = self:GhostCreate()
+		end
+		self.ACGhost:SetLocalPos(offset)
+		self.ACGhost:SetLocalAngles(Angle(0, 0, roll))
+	end
+end
 
 function ENT:OnRemove()
 	for _, ply in pairs(player.GetAll()) do
 		if ply:GetViewEntity() == self then
 			ply:SetViewEntity(nil)
 		end
 	end
+	
+	if IsValid(self.ACGhost) then
+		self.ACGhost:Remove()
+	end
```

Comment:
You don't need this. Instead, add `self:DeleteOnRemove(self.ptexture)` to `ENT:GhostCreate`. See https://github.com/NeatNit/hl_camera/blob/master/lua/entities/hl_camera/projection.lua#L89